provisioner/terraform/tfparse/tfextract.go

```diff
@@ -0,0 +1,183 @@
+package tfparse
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"slices"
+	"sort"
+	"strings"
+
+	"github.com/coder/coder/v2/provisionersdk/proto"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclparse"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/hashicorp/terraform-config-inspect/tfconfig"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+)
+
+// WorkspaceTags extracts tags from coder_workspace_tags data sources defined in module.
+func WorkspaceTags(ctx context.Context, logger slog.Logger, module *tfconfig.Module) (map[string]string, error) {
+	workspaceTags := map[string]string{}
+
+	for _, dataResource := range module.DataResources {
+		if dataResource.Type != "coder_workspace_tags" {
+			logger.Debug(ctx, "skip resource as it is not a coder_workspace_tags", "resource_name", dataResource.Name, "resource_type", dataResource.Type)
+			continue
+		}
+
+		var file *hcl.File
+		var diags hcl.Diagnostics
+		parser := hclparse.NewParser()
+
+		if !strings.HasSuffix(dataResource.Pos.Filename, ".tf") {
+			logger.Debug(ctx, "only .tf files can be parsed", "filename", dataResource.Pos.Filename)
+			continue
+		}
+		// We know in which HCL file is the data resource defined.
+		file, diags = parser.ParseHCLFile(dataResource.Pos.Filename)
+
+		if diags.HasErrors() {
```

Comment:
Should the line gap be here? In other places `if diags.HasErrors()` is immediately after the previous statement. As this is error handling I'd assume we don't want the gap.

```suggestion
		file, diags = parser.ParseHCLFile(dataResource.Pos.Filename)
		if diags.HasErrors() {
```